remove_wheels.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# fail on undefined variables
+set -u
+# Prevent pipe errors to be silenced
+set -o pipefail
+# Exit if any command exit as non-zero
+set -e
+# enable trace mode (print what it does)
+set -x
+
+# get the anaconda token from the github secrets
+#
+# this is to prevent accidental removals
+echo "Getting anaconda token from github secrets..."
+
+ANACONDA_USER="${INPUT_ANACONDA_USER}"
+ANACONDA_TOKEN="${INPUT_ANACONDA_TOKEN}"
+N_LATEST_UPLOADS="${INPUT_N_LATEST_UPLOADS}"
+
+
+# if the ANACONDA_TOKEN is empty, exit with status -1
+# this is to prevent accidental removals
+if [ -z "${ANACONDA_TOKEN}" ]; then
+  echo "ANACONDA_TOKEN is empty, exiting..."
+  exit -1
+fi
+
+# if the N_LATEST_UPLOADS is empty, exit with status -1
+# as this should be set in by the user and it is better
+# to fail on this to signal a problem. i.e.,
+# explicit is better than implicit.
+if [ -z "${N_LATEST_UPLOADS}" ]; then
+  echo "N_LATEST_UPLOADS is empty, exiting..."
+  exit -1
+fi
+
+
+# Query the package index for packages
+#
+# TODO: should be possible to alter this, since separating the workflow
+# into two steps, one for uploading and one for cleanup, should make it
+# possible for users to manually trigger the cleanup step before/after the
+# upload step has completed in their own repos instead of us having to do it.
+#
+# TODO: raises questions on how to moderate cleanups among multiple users
+# operating on the same channel, but that might be a different issue.
+curl https://raw.githubusercontent.com/scientific-python/upload-nightly-action/main/packages-ignore-from-cleanup.txt --output packages-ignore-from-cleanup.txt
```

Comment:
Please see above for context and my question on this. :)

Reply:
I've put some thought into it, and this can be included as an input in the action. There would be two cases:

- An API token for package X that can delete older wheels for X
- An API token for the organisation that can perform index-wide deletions of older wheels for packages

While the former would be recommended to limit deletion scopes, when the latter is used by the action users (as it is being done here for this repo), a `whitelist_packages:` input can be added to include a comma-separated list of packages (`"openblas-libs"` in our case).

Another reasonable message could be to add a warning: `"Wheels for package X requested for deletion, but X is whitelisted. Please either remove it from the whitelist or try to delete a different package as needed."`.